Scripts/package.py

```diff
@@ -234,6 +235,32 @@ def main():
         return
     os.makedirs(CACHE_DIRECTORY, exist_ok=True)
 
+    # Checking version.cs
```

Comment:
Could you put the code in line 238-262 in a separate function? That way a future reader has the option to skip reading that when reading main.

Also, with a function call you can make the effect of the `trail` parameter more explicit:
```
if not args.trail:
  check_version_cs()
```

Reply:
Done -- except that we want to check the version info in any case; --trial says to continue even if the info is mismatched